tools/src/main/scala/scala/scalanative/build/ScalaNative.scala

```diff
@@ -82,7 +82,9 @@ private[scalanative] object ScalaNative {
             linker.Link(config, linked.entries, optimized)
           }
         } else {
-          linked
+          config.logger.time("Optimizing (false)") {
```

Comment:
This was actually deliberate because part of the issue was it was initial hard to figure out the difference via the logs in CI. I can move this to another PR if desired. The output is shown first for different optimization levels:
```
[info] Linking (786 ms)
[info] Checking intermediate code (31 ms)
[info] Dumping intermediate code (linked) (243 ms)
[info] Discovered 622 classes and 4138 methods
[info] Optimizing (debug mode) (394 ms)
[info] Checking intermediate code (26 ms)
[info] Dumping intermediate code (optimized) (259 ms)
[info] Dumping intermediate code (lowered) (370 ms)
[info] Generating intermediate code (860 ms)
[info] Produced 8 files
[info] Compiling to native code (3518 ms)
[info] Linking native code (immix gc, none lto) (156 ms)
[info] Total (6293 ms)

```
The change with no optimize makes it pretty clear that it is a no-op (0 time). Without this change there is no "Optimizing" output.
```
[info] Linking (5035 ms)
[info] Checking intermediate code (931 ms)
[info] Dumping intermediate code (linked) (3595 ms)
[info] Discovered 5122 classes and 32001 methods
[info] Optimizing (false) (0 ms)
[info] Checking intermediate code (1101 ms)
[info] Dumping intermediate code (optimized) (3196 ms)
[info] Dumping intermediate code (lowered) (5061 ms)
[info] Generating intermediate code (15676 ms)
[info] Produced 8 files
[info] Compiling to native code (5942 ms)
[info] Linking native code (immix gc, none lto) (831 ms)
[info] Total (36493 ms)
```

Reply:
We could also add a `Mode.none` which would be like optimizer false. We could deprecate the `withOptimize` but I am not sure how that code would be refactored. We would see output like `Optimizing (none mode)` like we see `none lto`.

Reply:
Maybe for now we can change `Optimizing (false)` to `Optimizing ( skipped )` as it would be more verbose. 
We probably might think about limiting the scope of user available settings for the user. eg.` withOptimize` could stay but be only available in developers build.